sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/DoubleHistogramPointData.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.data;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.api.metrics.common.Labels;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * DoubleHistogramPointData represents an approximate representation of the distribution of
+ * measurements.
+ */
+@Immutable
+@AutoValue
+public abstract class DoubleHistogramPointData implements PointData {
+  /**
+   * Functional interface for consuming bucket boundaries and counts as a sequence of pair values.
+   */
+  public interface BucketConsumer {
+    void accept(double upperBound, long count);
+  }
+
+  /**
+   * Creates a DoubleHistogramPointData.
```

Comment:
Can you add documentation here about the relative sizes of the boundaries and counts? And, we should definitely have some validation on creation that the sizes are valid, I think.

Reply:
make sense, added.